doc/source/user/quickstart.rst

```diff
@@ -361,7 +361,7 @@ existing array rather than create a new one.
     >>> a += b                            # b is not automatically converted to integer type
     Traceback (most recent call last):
         ...
-    numpy.core._exceptions.UFuncTypeError: Cannot cast ufunc 'add' output from dtype('float64') to dtype('int64') with casting rule 'same_kind'
+    numpy.core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'add' output from dtype('float64') to dtype('int64') with casting rule 'same_kind'
```

Comment:
Hmm, @eric-wieser I think you had an opinion on the original issue? I had tried in ipython, and I thought it just printed the `__name__`, but maybe not.

I think I will just open an issue about it after this is done, but it would seem nice if we can find a solution and also print out a name that we want users to use in `try: ... except: Error`. `UFuncTypeError` seems good, but it should not be in a private module, and maybe we should just use a single class for all the subclasses (at least until we make them public), to avoid printing a private name?

Reply:
> maybe we should just use a single class for all the subclasses (at least until we make them public), to avoid printing a private name?

One of the ideas behind the subclasses was to avoid actually paying any string formatting costs until the traceback is printed. Merging them into one class would make that more difficult.

Reply:
OK, maybe the real question is different: If `UFuncTypeError` is private (due to being defined in `_exceptions`), should this set it to `TypeError.__qualname__` and `TypeError.__name__`, etc. instead?

It might not be hard to fix this differently by using `__reduce__(self): return f"{original_module}.{original_qualname}"`.

Reply:
Overloading `__reduce__` sounds promising to me

Reply:
I guess the question is whether right now we are OK with messing up the print slightly to fix the pickling. Sorry, I thought this was much clearer :(.

I am not sure `__reduce__` will work easily. It will be a solution for the error instance, but I doubt it will work for the error class.  Now the original bug report did not have an issue with that, so maybe we can get away with a class that cannot be pickled (but its instances can).